### PR TITLE
logical: Periodically send heartbeats to logical replication streams and retry ingestion on errors

### PR DESCRIPTION
**Pull request by navsetlur**
See commit description, but in summary the heartbeat now works for logical data replication and triggers a retry on a heartbeat failure.

### CODE REVIEW

File: pkg/ccl/crosscluster/logical/logical_replication_job_test.go [hunk 1]
```diff
@@ -581,6 +529,102 @@ func TestLogicalAutoReplan(t *testing.T) {
...
+
+	// Cancel the producer job and wait for the hearbeat to pick up that the stream is inactive
+	t.Logf("Canceling  replication producer %s", prodAID)
+	dbA.QueryRow(t, "CANCEL JOB $1", prodAID)
+
+	// The ingestion job should eventually retry because it detects 2 nodes are dead
+	require.ErrorContains(t, <-retryErrorChan, fmt.Sprintf("replication stream %s is not running, status is STREAM_INACTIVE", prodAID))
+}
+
+func setupLogicalTestServer(
```
Comment (stevendanna) [inline pkg/ccl/crosscluster/logical/logical_replication_job_test.go:595]: :D I think I have this same function on another PR. Nice cleanup.

**Review by stevendanna — COMMENTED** [review]
Thanks for working on this.

Minor nit in the commit message:

`Periodically -> Periodically`

I think we may want to go a little further here and add something like the physical replication job's `waitUntilProducerActive` so that we actually move into a `PAUSED` state if the user cancel's the producer side job. This gives the user the ability to force the replication process to stop from the source.

**Review by stevendanna — APPROVED** [review]
Actually starting the heartbeater is a clear improvement, so also feel free to take my suggestion as a follow up.

**Comment by navsetlur** [comment]
bors r=stevendanna

**Pull request merged**